File: actions/HTML Encode.lbaction/Contents/Scripts/default.php
#!/usr/bin/env php
<?php

$items = array();

array_shift($argv);
foreach ($argv as $arg) {
    $item = array();
    $item['title'] = htmlentities($arg, ENT_QUOTES);
    $item['actionArgument'] = $item['title'];
    $items[] = $item;
}

echo json_encode($items);

File: Encode-Decode.lbext/Contents/Resources/Actions/Encode File as Data URI.lbaction/Contents/Scripts/default.py
#!/usr/bin/env python

import base64
import mimetypes
import os
import sys
import json

class FileNotFoundError(Exception):
    pass

def img_to_data(path):
    """Convert a file (specified by a path) into a data URI."""
    if not os.path.exists(path):
        raise FileNotFoundError
    mime, _ = mimetypes.guess_type(path)
    with open(path, 'rb') as fp:
        data = fp.read()
        data64 = u''.join(base64.encodestring(data).splitlines())
        return u'data:%s;base64,%s' % (mime, data64)

items = []
for argv in sys.argv[1:]:
    try:
        data = img_to_data(argv)
        item = {}
        item['title'] = data
        item['subtitle'] = argv
        item['actionArgument'] = data
        item['quickLookURL'] = 'file://'+argv

        items.append(item)
    except FileNotFoundError:
        items.append({"title": "File not found."});

print(json.dumps(items))
